File: src/crypto/keys/PrivateKey.ts
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

 import PublicKey from './PublicKey';
import JsonWebKey from './JsonWebKey';

/**
 * Represents a Private Key in JWK format.
 * @class
 * @abstract
 * @hideconstructor
 */
export default abstract class PrivateKey extends JsonWebKey {

  /**
   * Default Sign Algorithm for JWK 'alg' field
   */
  readonly alg: string = 'none';

  /**
   * Gets the corresponding public key
   * @returns The corresponding {@link PublicKey}
   */
  abstract getPublicKey (): PublicKey;
}

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/keys/PrivateKey.kt
package com.microsoft.did.sdk.crypto.keys

import com.microsoft.did.sdk.crypto.models.webCryptoApi.JsonWebKey

abstract class PrivateKey(key: JsonWebKey) : PublicKey(key) {
    /**
     * Default Sign Algorithm for JWK 'alg' field
     */
    override var alg: String? = if (key.alg != null) key.alg else "none"

    /**
     * Gets the corresponding public key
     * @returns The corresponding {@link PublicKey}
     */
    abstract fun getPublicKey(): PublicKey

    override fun minimumAlphabeticJwk(): String {
        return this.getPublicKey().minimumAlphabeticJwk()
    }
}